satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,554 @@
+     - a) account delinquent.
+     - b) illegal content.
+     - c) Malicious Links.
+     - d) other.
+   - Unauthorized users should not see this button or drop down.
+2. I want the account details to show that it's obvious if a user is suspended so that they can
+   easily be manually unsuspended if needed.
+
+   Acceptance criteria:
+   - The UX makes it quick to unsuspend a user once suspended
+
+__When an account needs to be unsuspended/re-activated__
+
+I want authorized users to see a simple "reactivate" button in the user's detail view.
+
+Acceptance criteria:
+- Button resumes a user's limits to what they had them set at before suspension occurred.
+- Add an optional admin note of the reason for unsuspending.
+
+__When an account needs a limit increase or decrease__
+
+I want authorized users to be able to input the allotted usage for:
+- Storage limit, and select GB or TB.
+- Egress limit, and select GB or TB
+- Segment limit, and enter the max number with formatted numerical input fields for better
+  readability.
+- Bucket limit, and enter the number of buckets.
+
+__When an account needs to be deleted__
+
+I want authorized users to be able to press a "delete account" button along with a "confirm" delete
+button that confirms this user and all their information and stored data are to be deleted, so that
+we don't have to manually access their API keys and manually delete their data.
+
+__When an account needs to be created__
+
+I want authorized users to have an option in their initial login view to create a user.
+
+This operation rarely happens so the design of the UX shouldn’t make it prominent
+
+Acceptance criteria:
+- Input fields for: full name, email, generate (temporary) password.
+
+__When an account needs more projects__
+
+I want to locate the user I need to enable a higher project allowance for and have an option to
+increase the number of projects the account is allowed to have in the account details view.
+
+Acceptance criteria:
+- Field to increase or decrease the account's project count.
+- Project limits default to the user's project limits or can be set if otherwise.
+
+__When a user needs to be added to a different user's project__
+
+I want to be able to search for the project ID.
+I need to add the user to, so that I can view the members off that project and add and/or remove a
+user.
+
+__When a user needs their multi factor authentication (MFA) reset__
+
+I want there to be a MFA reset option in the account details view.
+
+Acceptance criteria:
+- We can see that MFA is enabled or disabled.
+- We can only disable MFA.
+
+__When a user wants to change the email address tied to their account__
+
+I want to be able to edit the email address in the account details view so that all events and data
+associated to that user is now under the updated email address.
+
+##### Audit trail
+
+The back office keeps a register of all performed operations and who executed them that cause
+modifications.
```

Comment:
We probably should keep track of `reading` data as well, not just modification.

Reply:
That wasn't specified in the requirements document https://docs.google.com/document/d/1KzZuhfvyU1CgrQm9L4moYxtQME_V6zTYuc6pdarKrUk/edit#heading=h.a7bdzb7w5g2a

What is it going to provide? This audit trail is shown on certain pages (e.g. see at the bottom of https://storj.github.io/admin-ui/account-details)

I understand that we may want to log/audit the user access to information, but we could keep that in a log rather than in the DB

Reply:
Ah, yes, that makes sense. I just want to make sure that reading information also needs some sort of audit trail.

Reply:
I will rename it.
Do you think that we should also have the audit trail in a log format in this version or we should write it in the _out of scope_ section?

Reply:
Audit trail should definitely part of the initial version.

Reply:
I'll add that audit trail to log to a file using a specific logger only for it.